packages/tools/client-debugger/client-debugger-view/src/components/TelemetryView.tsx

```diff
@@ -64,6 +69,37 @@ export function TelemetryView(): React.ReactElement {
 		ITimestampedTelemetryEvent[] | undefined
 	>();
 	const [maxEventsToDisplay, setMaxEventsToDisplay] = React.useState<number>(DEFAULT_PAGE_SIZE);
+	const [selectedCategory, setSelectedCategory] = useState("");
+	const [filteredTelemetryEvents, setFilteredTelemetryEvents] = React.useState<
+		ITimestampedTelemetryEvent[] | undefined
+	>();
+	const [customSearch, setCustomSearch] = React.useState("");
+	const [eventNameOptions, setEventNameOptions] = useState<string[]>([]);
+	const [matchingOptions, setMatchingOptions] = React.useState([...eventNameOptions]);
```

Comment:
Initializing your state to another piece of state likely won't work out how you want. Probably best to initialize this to empty or undefined, and update it alongside the other state within the effect.

Reply:
Yep, The current code would initialize the default value of matchingOptions to eventNameOptions (which is initialized to an empty array).